pkg/client/unversioned/request.go

```diff
@@ -425,7 +425,36 @@ func (r *Request) VersionedParams(obj runtime.Object, convertor runtime.ObjectCo
 	}
 	for k, v := range params {
 		for _, vv := range v {
-			r.setParam(k, vv)
+			// TODO: Move it to setParam method, once we get rid of
+			// FieldSelectorParam & LabelSelectorParam methods.
+			value := vv
+			if k == unversioned.LabelSelectorQueryParam(r.apiVersion) && value == "" {
+				// Don't set an empty selector for backward compatibility.
+				continue
+			}
+			if k == unversioned.FieldSelectorQueryParam(r.apiVersion) {
+				if value == "" {
+					// Don't set an empty selector for backward compatibility.
+					continue
+				}
+				// TODO: Filtering should be handled somewhere else.
+				selector, err := fields.ParseSelector(value)
+				if err != nil {
+					r.err = err
```

Comment:
It might be good to expand this, e.g. `r.err = fmt.Errorf("unparsable field selector: %v", err)`, so people can tell where the error came from more easily.

Same below.